src/com/lftechnology/java/training/dipak/employeemanagement/controller/EmployeeController.java

```diff
@@ -0,0 +1,221 @@
+
+            ed.terminateEmployee(em);
+
+        } catch (Exception ex) {
+            LOGGER.log(Level.INFO, exceptionOccurred, ex);
+        }
+        return count;
+    }
+
+    /**
+     * <p>
+     * This method takes the input for the filters and calls the viewEmployee method.
+     * </p>
+     * 
+     * @param e
+     * @param sc
+     * @return
+     */
+    public ResultSet viewEmployee(Employee empl, Scanner sc) {
+
+        String fullName = null;
+        String department = null;
+        String address = null;
+        try {
+            LOGGER.info("Please enter the employee fullname::");
+            fullName = sc.nextLine();
+
+            LOGGER.info("Please enter the department::");
+            department = sc.nextLine();
+
+            LOGGER.info("Please enter the address::");
+            address = sc.nextLine();
+
+            empl.setFullName(fullName);
+            empl.setAddress(address);
+            empl.setDepartment(department);
+
+            EmployeeService ed = ServiceFactory.getEmployeeService();
+
+            ed.viewEmployee(empl);
+
+        } catch (Exception ex) {
+            LOGGER.log(Level.INFO, exceptionOccurred, ex);
+        }
+        return null;
+    }
+
+    /**
+     * <p>
+     * This method takes the input for the fields to be edited and calls editEmployeeDetails in service layer.
+     * </p>
+     * 
+     * @param e
+     * @param sc
+     * @return e
+     */
+    public Employee editEmployeeDetails(Employee emp1, Scanner sc) {
+
+        LOGGER.info("Welcome to the edit mode....");
+        String userName = null;
+        String password = null;
+        String fullName = null;
+        String department = null;
+        String address = null;
+        Console cnsl = null;
+        try {
+            cnsl = System.console();
+            LOGGER.info("Enter new username::");
+            userName = sc.nextLine();
+
+            LOGGER.info("Enter new password::");
+            char[] pwd = cnsl.readPassword();
+            password = String.valueOf(pwd);
+
+            LOGGER.info("Enter new fullname::");
+            fullName = sc.nextLine();
+
+            LOGGER.info("Enter new department::");
+            department = sc.nextLine();
+
+            LOGGER.info("Enter new address::");
+            address = sc.nextLine();
+
+            if (!("".equals(userName))) {
+                emp1.setUserName(userName);
+            }
+            if (!("".equals(password))) {
+                emp1.setPassword(password);
+            }
+            if (!("".equals(fullName))) {
+                emp1.setFullName(fullName);
+            }
+            if (!("".equals(department))) {
+                emp1.setDepartment(department);
+            }
+            if (!("".equals(address))) {
+                emp1.setAddress(address);
+            }
+
+            EmployeeService es = ServiceFactory.getEmployeeService();
+
+            emp1 = es.editEmployeeDetails(emp1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "emp1". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)